Kernel/Memory/TypedMapping.h

```diff
@@ -65,4 +94,51 @@ static ErrorOr<TypedMapping<T>> map_typed_writable(PhysicalAddress paddr)
     return map_typed<T>(paddr, sizeof(T), Region::Access::Read | Region::Access::Write);
 }
 
+template<typename T>
+static ErrorOr<NonnullOwnPtr<TypedMapping<T[]>>> adopt_new_nonnull_own_typed_mapping_array(PhysicalAddress paddr, size_t items, Region::Access access = Region::Access::Read)
+{
+    size_t length_in_bytes = items * sizeof(T);
+    auto mapping_length = TRY(page_round_up(paddr.offset_in_page() + length_in_bytes));
+    auto region = TRY(MM.allocate_mmio_kernel_region(paddr.page_base(), mapping_length, {}, access));
+    auto table = TRY(adopt_nonnull_own_or_enomem(new (nothrow) Memory::TypedMapping<T[]>()));
+    table->region = move(region);
+    table->offset = paddr.offset_in_page();
+    table->paddr = paddr;
+    table->length = length_in_bytes;
+    return table;
+}
+
+template<typename T>
+static ErrorOr<TypedMapping<T[]>> map_typed_array(PhysicalAddress paddr, size_t items, Region::Access access = Region::Access::Read)
+{
+    size_t length_in_bytes = items * sizeof(T);
+    return map_typed<T[]>(paddr, length_in_bytes, access);
+}
+
+template<typename T>
+static ErrorOr<TypedMapping<T[]>> map_typed_array(PhysicalAddress paddr, size_t items)
```

Comment:
Shouldn't this overload be unnecessary? The `access` parameter of the other `map_typed_array` implementation has a default value